.github/workflows/unit_tests.yml

```diff
@@ -41,8 +41,15 @@ jobs:
         pip install gcovr==7.2
     - name: coverage
       if: contains(matrix.os, 'ubuntu')
-      run:
-        gcovr -r ./src --html-details ./coverage/report.html --html-theme github.dark-green build/test
+      run: |
+        gcovr --html-details ./coverage/report.html \
+              --html-theme github.dark-green \
+              --filter bcmp \
+              --filter middleware \
+              --filter dirvers \
```

Comment:
drivers? 😄  Have you looked at a coverage report to see whether this is working?

Reply:
I'm able to generate reports locally, using this command: `gcovr --html-details ./build/coverage/report.html --html-theme green --filter bcmp --filter common --filter drivers --filter middleware --filter network build/test/test`
<img width="1422" alt="Screenshot 2024-11-04 at 10 13 24 AM" src="https://github.com/user-attachments/assets/55a89d1d-265b-4c06-b99b-02313cc30adb">

I haven't made any changes to the test/cmakelists.txt so I wonder why that is happening. But I'm also not sure if this is a complete report.